### PR TITLE
Create challenge api

### PR DESCRIPTION
**Pull request by Shehanx86**

### CODE REVIEW

File: services/challenge.service.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { readFile } from "fs/promises";
+import {
+  deleteChallengeById,
+  fetchChallengeById,
+  fetchChallengesByUserID,
+  saveChallenge,
+  updateChallengeById,
+} from "../repository/challenge.repository.js";
```
Comment (dulshan-perera) [inline services/challenge.service.js]: Import from the index file

File: repository/challenge.repository.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export const fetchChallengeById = (id) =>
+Challenge.findById(id)
+
+// Update challenge repository logic
+export const updateChallengeById = (id, data) =>
+Challenge.findByIdAndUpdate(id, data, { new: true })
+
+// Delete challenge repository logic
+export const deleteChallengeById = (id) =>
+  Challenge.findByIdAndDelete(id)
```
Comment (dulshan-perera) [inline repository/challenge.repository.js:21]: import and export in the repository index.js file

**Review by dulshan-perera — CHANGES_REQUESTED** [review]
Check comment and update

File: services/challenge.service.js (hunk 1 above)
Comment (Shehanx86) [reply services/challenge.service.js]: done

File: repository/challenge.repository.js (hunk 2 above)
Comment (Shehanx86) [reply repository/challenge.repository.js:21]: done

**Review by dulshan-perera — APPROVED** [review]
LGTM

**Pull request merged**